### PR TITLE
HPCC-14495 ApplicationVariables filterable in workunitservices.workunitlist

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -2889,9 +2889,12 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                     namefilter.set(fv);
                 else if (subfmt==WUSFappvalue)
                 {
-                    query.append("[Application/").append(fv).append("=?~\"");
+                    const char *app = fv;
                     fv = fv + strlen(fv)+1;
-                    query.append(fv).append("\"]");
+                    query.append("[Application/").append(app);
+                    if (fv && *fv)
```
Comment (ghalliday) [inline common/workunit/workunit.cpp]: minor: test for fv is redundant.  

The embedded null character isn't a good interface, but it looks like that's what it did before.

**Comment by ghalliday** [comment]
Should you also increase the eclhelper minimum activity version - since this will make existing workunits incompatible.

**Comment by ghalliday** [comment]
@richardkchapman two comments

**Comment by richardkchapman** [comment]
I think existing workunits will fail to load...

We should increase the activity version before 6.0.0 goes gold (or indeed rc1) but not sure if we should on every change...

**Comment by richardkchapman** [comment]
@ghalliday repushed to address comments

**Pull request merged**